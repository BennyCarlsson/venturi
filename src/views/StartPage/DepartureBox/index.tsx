import Heading, { HeadingType } from 'components/Heading';
import Paragraph from 'components/Paragraph';
import colors from 'tokens/colors';
import { FontWeight } from 'types';
import {
  BottomPart,
  BottomParagraph,
  BottomLeftPart,
  BottomRightPart,
  DepartsWrapper,
  NextDepartureWrapper,
  StyledTrailLogo,
  TopPart
} from './styles';

type NextDepartureBoxProps = {
  name: string | undefined;
  track: string | undefined;
};

const DepartureBox = (props: NextDepartureBoxProps) => {
  const { name, track } = props;
  return (
    <NextDepartureWrapper>
      <TopPart>
        <Heading
          headingType={HeadingType.H1}
          fontWeight={FontWeight.medium}
          fontSize={18}
          color={colors.white}
        >
          Departs from
        </Heading>
        <DepartsWrapper>
          {/* Todo: icon after type */}
          <StyledTrailLogo />
          <Heading
            headingType={HeadingType.H2}
            fontWeight={FontWeight.medium}
            fontSize={28}
            color={colors.white}
          >
            {name}
          </Heading>
        </DepartsWrapper>
      </TopPart>
      <BottomPart>
        <BottomLeftPart>
          <Paragraph fontWeight={FontWeight.medium} fontSize={14} color={colors.white}>
            Platform
          </Paragraph>
          <BottomParagraph fontWeight={FontWeight.medium} fontSize={24} color={colors.white}>
            {track}
          </BottomParagraph>
        </BottomLeftPart>
        <BottomRightPart>
          <Paragraph fontWeight={FontWeight.medium} fontSize={14} color={colors.white}>
            Destination name
          </Paragraph>
          <BottomParagraph fontWeight={FontWeight.medium} fontSize={24} color={colors.white}>
            Placeholder
          </BottomParagraph>
        </BottomRightPart>
      </BottomPart>
    </NextDepartureWrapper>
  );
};

export default DepartureBox;
