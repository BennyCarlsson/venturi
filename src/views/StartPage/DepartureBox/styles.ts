import styled from "styled-components";
import Colors from "tokens/colors";
import { ReactComponent as TrainLogo } from "assets/icons/train.svg";
import { motion } from "framer-motion";
import MotionParagraph from "components/MotionParagraph";

export const NextDepartureWrapper = styled(motion.div)`
  border-radius: 0 0 8px 8px;
  background-color: ${Colors.vBlue};
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.1);
`;

export const TopPart = styled(motion.div)`
  padding: 24px;
  border-bottom: 1px solid ${Colors.borderWhite};
`;

export const BottomPart = styled.div`
  display: flex;
`;

export const BottomLeftPart = styled(motion.div)`
  padding: 16px 24px;
  width: 100%;
  border-right: 1px solid ${Colors.borderWhite};
`;

export const BottomRightPart = styled(motion.div)`
  padding: 16px 24px;
  width: 100%;
`;

export const BottomParagraph = styled(MotionParagraph)`
  padding-top: 8px;
`;

export const StyledTrailLogo = styled(TrainLogo)`
  width: 32px;
  height: 32px;
  margin-right: 16px;
  path {
    fill: ${Colors.white};
  }
`;

export const DepartsWrapper = styled.div`
  display: flex;
  padding-top: 8px;
`;
