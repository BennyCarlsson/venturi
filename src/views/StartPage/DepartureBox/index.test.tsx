import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import DepartureBox from '.'

describe('<DepartureBox />', () => {
  const name = 'Test name'
  const track = 'Test Track'

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const tree = renderer
        .create(<DepartureBox name={name} track={track} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('Component', () => {
    beforeEach(() => {
      render(<DepartureBox name={name} track={track} />)
    })

    it('should render without crashing', () => {
      expect(screen).toBeTruthy()
    })

    it('should render name and track props', () => {
      expect(screen.getByText(name)).toBeInTheDocument()
      expect(screen.getByText(track)).toBeInTheDocument()
    })
  })
})
