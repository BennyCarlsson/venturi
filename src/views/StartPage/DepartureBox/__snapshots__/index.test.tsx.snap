// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DepartureBox /> snapshots should match snapshot 1`] = `
<div
  className="sc-iBPRYJ HCnjh"
  style={Object {}}
>
  <div
    className="sc-fubCfw hIKVhW"
    style={Object {}}
  >
    <span
      style={Object {}}
    >
      <h1
        className="sc-bdfBwQ UZuHf"
        color="#FFFFFF"
        fontSize={18}
        fontWeight={500}
        style={Object {}}
      >
        Departs from
      </h1>
    </span>
    <div
      className="sc-dQppl dWSjyx"
    >
      <svg
        className="sc-crrsfI gWLlXM"
      >
        train.svg
      </svg>
      <h2
        className="sc-gsTCUz htLLms"
        color="#FFFFFF"
        fontSize={28}
        fontWeight={500}
      />
    </div>
  </div>
  <div
    className="sc-pFZIQ hxExeM"
  >
    <div
      className="sc-jrAGrp jzTLuk"
      style={Object {}}
    >
      <p
        className="sc-gKsewC bMoIAp"
        color="#FFFFFF"
        fontSize={14}
        fontWeight={500}
        style={Object {}}
      >
        Platform
      </p>
      <p
        className="sc-gKsewC iTDTZO sc-iqHYGH jyHRlY"
        color="#FFFFFF"
        fontSize={24}
        fontWeight={500}
        style={Object {}}
      >
        Test Track
      </p>
    </div>
    <div
      className="sc-kEjbxe bnNZBb"
      style={Object {}}
    >
      <p
        className="sc-gKsewC bMoIAp"
        color="#FFFFFF"
        fontSize={14}
        fontWeight={500}
        style={Object {}}
      >
        Destination name
      </p>
      <p
        className="sc-gKsewC iTDTZO sc-iqHYGH jyHRlY"
        color="#FFFFFF"
        fontSize={24}
        fontWeight={500}
        style={Object {}}
      >
        Test name
      </p>
    </div>
  </div>
</div>
`;
